.github/workflows/checks.yml

```diff
@@ -108,6 +95,19 @@ jobs:
         id: changeset-status
         uses: ./.github/actions/changeset-status
 
+      - name: Check if Version Already Exists
+        if: steps.changeset-status.outputs.new_version
+        run: |
+          NAME="${{ steps.changeset-status.outputs.name }}"
+          NEW_VERSION="${{ steps.changeset-status.outputs.new_version }}"
+
+          # Check that a tag for the new version does not already exist
+          if [[ ! -z $(git tag -l "$NAME@$NEW_VERSION") ]]; then
+            echo "🚨 The version '$NAME@$NEW_VERSION' already exists."
+            echo "🚨 Please ensure that the version bump and the base branch is correct."
+            exit 1
+          fi
```

Comment:
protects against accidental release of an existing version to avoid such cases https://github.com/StackVista/rancher-extension-stackstate/pull/58

the base branch was created from a wrong commit, that included a major changeset

here is an example job failed in the fork https://github.com/owanturist/rancher-extension-stackstate/actions/runs/12673427943/job/35319785896